homeassistant/components/nuki/lock.py

```diff
@@ -65,15 +70,15 @@ def service_handler(service):
             if service.service == SERVICE_LOCK_N_GO:
                 unlatch = service.data[ATTR_UNLATCH]
                 lock.lock_n_go(unlatch=unlatch)
-            elif service.service == SERVICE_UNLATCH:
-                lock.unlatch()
+            elif service.service == SERVICE_CHECK_CONNECTION:
+                lock.check_connection()
 
-    hass.services.register(
-        DOMAIN, SERVICE_LOCK_N_GO, service_handler,
+    hass.services.async_register(
```

Comment:
We're in sync context so we have to use the home assistant sync api.
```suggestion
    hass.services.register(
```

Reply:
Done